### DISCUSSION TITLE
Converting a regular vite react app to SSR

### DISCUSSION THREAD

**Discussion by iBuaSpaz** [vikejs/vike#620] [Discussion]
So i installed this on my current Vite react app, why is it none of these plugins just works out of the box.

Its called Vite-plugin-ssr yet you have to change your intire Vite code to make this work.

First off, my files are in src dir, they are named Jsx files, already now there is a issue.

Second, i need to add and change a bunch of files, and there is just no simple approch for this, can we all agree you make a website in PHP, it just works?! i finally found out how to use vite react to make a nice looking webpage... BAM.. no SEO, you now need to learn SSR.

I did programs in PHP, ASP, C#, C++, Flutter ect, i have never in my life been strugling so hard with making a single website, and im concerned i have to spend a week just setting up the base for next site, i mean, is there no simple way to add SSR to a existing vite react site, like install a plugin, add 2-3 js files, and your running, no need to change anything else? this feel very hack'n'slash cut and hope something works approch.

Im very open minded, but this is more confusing then any programming language i have ever used, there just isnt anything simple in it..

**Comment by samuelstroschein** [comment]
the maintainer is working on a "single config file" design vor vite-plugin-ssr https://github.com/brillout/vite-plugin-ssr/issues/578. 

that said, i love your sentiment toward JS. it's a dumpster fire

**Reply by brillout** [reply]
Tbh, VPS is not particularly friendly to newcomers that merely want to ship a product and don't care about technicalities. For such use cases, (upcoming) VPS higher-level frameworks are the way to go.

**Comment by brillout** [comment]
VPS is more tailored towards folks who are well versed with the JS world and who want/need/like control.

I understand your frustration, and I'm woking on a solution that, I think, will delight JS newcomers but I'm not there yet. In the meantime use one of these other higher level frameworks.

**Reply by iBuaSpaz** [reply]
> VPS is more tailored towards folks who are well versed with the JS world and who want/need/like control.
> 
> I understand your frustration, and I'm woking on a solution that, I think, will delight JS newcomers but I'm not there yet. In the meantime use one of these other higher level frameworks.

I have been working on MMO emulators in JS, what is frustrating is, you make a site, now, you have to totally rewrite this site so SEO works, and the "programming" of this page is so different from regular vite React pages.

Same if you have a car where every tire has 2 different size nuts, would not make much sense..

I really look forward to a system thats more user friendly then the current, because the current system is really bad imho.

**Reply by brillout** [reply]
Agreed.

**Reply by samuelstroschein** [reply]
@iBuaSpaz do you have suggestions on making VPS more user-friendly?
